pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -440,18 +435,16 @@ func (nc *Controller) Run(stopCh <-chan struct{}) {
 		go nc.taintManager.Run(stopCh)
 	}
 
-	if nc.taintNodeByCondition {
```

Comment:
@k82cn I moved the condition to inside the worker (line 481) since we do want to reconcile node labels even if taintNodeByCondition is false.